.github/workflows/ci-test.yml

```diff
@@ -7,35 +7,103 @@ on:
     branches: [main, master, release-*, gha-test-*]
 
 jobs:
-  #dev-setup-sh-test:
-  #  runs-on: ubuntu-latest-xl
-  #  continue-on-error: false
-  #  strategy:
-  #    matrix:
-  #      target_os: [alpine, arch, centos, github]
-  #  steps:
-  #    - uses: actions/checkout@v2
-  #    - name: build image with dev-setup.sh
-  #      run: docker build -f docker/ci/${{ matrix.target_os }}/Dockerfile -t libra/build_environment:test .
+  prepare:
+    runs-on: ubuntu-latest
+    continue-on-error: false
+    outputs:
+      changes-target-branch: ${{ steps.changes.outputs.changes-target-branch }}
+      changes-base-githash: ${{ steps.changes.outputs.changes-base-githash }}
+      changes-pull-request-number: ${{ steps.changes.outputs.changes-pull-request-number }}
+      test-rust: ${{ steps.rust-changes.outputs.changes-found }}
+      test-dev-setup: ${{ steps.dev-setup-sh-changes.outputs.changes-found }}
+      test-website-build: ${{ steps.website-changes.outputs.changes-found }}
+      test-non-rust-lint: ${{ steps.non-rust-lint-changes.outputs.changes-found }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 #get all the history!!!
+      - id: changes
+        name: determine changes
+        uses: ./.github/actions/changes
+        with:
+          workflow-file: ci-test.yml
+      - id: rust-changes
+        name: find rust/cargo changes.
+        uses: ./.github/actions/matches
+        with:
+          pattern: '**.rs\|Cargo.toml\|Cargo.lock'
```

Comment:
I'm kinda of concerned about this as I mentioned elsewhere -- this is missing a lot of different files that affect the Rust build. Could we at least unconditionally use the determinator for Rust files for now?

Reply:
It was updated:   I agree it was risky before.

        with:
          pattern: '^documentation\|^docker\|^scripts'
          invert: "true"